### PR TITLE
Test enhancement

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log

- add other test.
- add the ```suggest``` key in ```composer.json``` to suggest the extensions.

### CODE REVIEW

**Comment by BenMorel** [comment]
`suggest` has already been proposed (and declined) in #5, see the reasons there!

File: tests/NativeCalculatorTest.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Brick\Math\Tests;
```
Comment (BenMorel) [inline tests/NativeCalculatorTest.php]: Should be Brick\Math\Tests\Internal\Calculator and file moved to the appropriate directory

File: tests/BigIntegerTest.php [hunk 2]
```diff
@@ -1898,6 +1898,15 @@ public function testSerialize()
         $this->assertBigIntegerEquals($value, unserialize(serialize($number)));
     }
 
+    public function testJsonSerialize()
```
Comment (BenMorel) [inline tests/BigIntegerTest.php]: Please move this test below testDirectCallToUnserialize!

**Comment by BenMorel** [comment]
Note to self: these lines [1](https://coveralls.io/builds/15757994/source?filename=src%2FInternal%2FCalculator%2FNativeCalculator.php#L60), [2](https://coveralls.io/builds/15757994/source?filename=src%2FInternal%2FCalculator%2FNativeCalculator.php#L104), [3](https://coveralls.io/builds/15757994/source?filename=src%2FInternal%2FCalculator%2FNativeCalculator.php#L201) can never be executed currently, due to higher-level optimizations such as [1](https://github.com/brick/math/blob/master/src/BigInteger.php#L195-L197), [2](https://github.com/brick/math/blob/master/src/BigInteger.php#L239-L241), [3](https://github.com/brick/math/blob/master/src/BigInteger.php#L287-L289). NativeCalculatorTest is needed if we want to cover them.

**Comment by peter279k** [comment]
Hi @BenMorel, thank you for your reply.
I think we have the ```NativeCalclator``` exist because I want to cover them.
And we should check that uncovered whether it's worked successfully.

File: tests/internal/NativeCalculatorTest.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
+<?php
```
Comment (BenMorel) [inline tests/internal/NativeCalculatorTest.php]: File name should be tests/Internal/Calculator/NativeCalculatorTest.php

**Comment by BenMorel** [comment]
I know, hence the "note to self" for posterity. Please just move the NativeCalculatorTest as requested and it will be good to merge!

**Comment by peter279k** [comment]
I'm sorry for that delay because I'm busy in the past of two days.

File: tests/Internal/Calculator/NativeCalculatorTest.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+ * Unit tests for class NativeCalculator.
+ */
+class NativeCalculatorTest extends AbstractTestCase
+{
+    /**
+     * @dataProvider providerAdd
+     *
+     * @param string $a               The input numerator.
+     * @param string $b               The input denominator.
+     * @param string $expectedValue   The expected denominator.
```
Comment (BenMorel) [inline tests/Internal/Calculator/NativeCalculatorTest.php]: It's not numerator & denominator here. You can just remove the descriptions.

File: tests/Internal/Calculator/NativeCalculatorTest.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            ['1234567891234567889999999', '1234567891234567889999999', '2469135782469135779999998'],
+        ];
+    }
+
+    /**
+     * @dataProvider providerMul
+     *
+     * @param string $a               The input numerator.
+     * @param string $b               The input denominator.
+     * @param string $expectedValue   The expected denominator.
```
Comment (BenMorel) [inline tests/Internal/Calculator/NativeCalculatorTest.php]: Same.

File: tests/Internal/Calculator/NativeCalculatorTest.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            ['1234567891234567889999999', '1234567891234567889999999', '1524157878067367851562259605883269630864220000001'],
+        ];
+    }
+
+    /**
+     * @dataProvider providerPow
+     *
+     * @param string $a               The input numerator.
+     * @param string $b               The input denominator.
+     * @param string $expectedValue   The expected denominator.
```
Comment (BenMorel) [inline tests/Internal/Calculator/NativeCalculatorTest.php]: Same.

**Comment by BenMorel** [comment]
All good, thanks!

**Pull request merged**